### DISCUSSION TITLE
Last invoice number in Stripe statement descriptor?

### DISCUSSION THREAD

**Discussion by ateuber** [pretix/pretix#2848] [Ideas]
Can we put the last invoice number in the Stripe statement descriptor?
It seems like this would ease the workflow for our secretary.
I think the relevant code is here: https://github.com/pretix/pretix/blob/master/src/pretix/plugins/stripe/payment.py#L380
But I think this should probably be a setting and also I'm not entirely sure it is a good idea. Any opinions?

**Comment by raphaelm** [comment]
The main problem with this is that statement descriptors are very limited in length and the limit varies by payment method.

Would it also easy your workflow to put it into the meta data we send to Stripe? This should also be accessible through Stripe's exports

**Reply by ateuber** [reply]
Unfortunately it needs to be on the receipt. Our bookkeeping keeps invoices and receipts together, sorted by invoice number. When taking the receipt from Stripe we need to look up the invoice number belonging to the order id.

**Reply by raphaelm** [reply]
What exactly do you mean by "receipt" here?

**Reply by ateuber** [reply]
This: https://stripe.com/docs/receipts
There is an example here: https://stripe.com/docs/terminal/features/receipts#prebuilt

**Reply by raphaelm** [reply]
That's what I guessed, and that's what confused me, usually these receipts are not used at all when working with pretix. See also their docs you just linked:

> When using [Checkout](https://stripe.com/docs/payments/checkout), you can automatically email your customers upon successful payments

We don't use checkout.

> When [creating a PaymentIntent](https://stripe.com/docs/payments/payment-intents), include your customer’s email address as a value for the [receipt_email](https://stripe.com/docs/api/payment_intents/create#create_payment_intent-receipt_email) parameter

We don't set that parameter.

> Stripe creates receipts when a customer pays an [invoice](https://stripe.com/docs/api/invoices), or makes any [subscription](https://stripe.com/docs/billing/subscriptions/creating) payment

We don't use invoice or subscription.

Thus, Stripe receipts never get sent when using pretix, and I don't think it makes much sense to use them for any bookkeeping purposes.

In the Stripe dashboard, I don't eben see an export of receipts? I only see an export on payments. How does your bookeeping department get data from stripe exactly?

**Reply by ateuber** [reply]
Sorry for causing confusion. Our bookkeeping just collects these receipts internally, like they do e.g. with bank transfer statements. They are not sent to customers. They download it by opening individual payments in Stripe and then in `receipt history` click on `view receipt` on the right. After clicking on `View it in your browser` it is printed as PDF. Don't ask me how necessary this work actually is. :roll_eyes:
